### ISSUE TITLE
BBC: Blead Breaks Sub::Quote

### ISSUE THREAD

**Issue by cjg-cguevara** [Perl/perl5#17491]
This is a bug report for perl from "Carlos Guevara" <carlos@carlosguevara.com>,
generated with the help of perlbug 1.41 running under perl 5.31.9.

-----------------------------------------------------------------
[Please describe your issue here]

BBC: Blead Breaks Sub::Quote

Please see http://matrix.cpantesters.org/?dist=Sub-Quote
```
[Please do not change anything below this line]
-----------------------------------------------------------------
---
Flags:
    category=core
    severity=low
---
Site configuration information for perl 5.31.9:

Configured by cpan at Sun Jan 26 08:54:13 CST 2020.

Summary of my perl5 (revision 5 version 31 subversion 9) configuration:
  Commit id: fd9c22e56536c51f0c621c365b440c3ca01c1ce4
  Platform:
    osname=freebsd
    osvers=12.1-release-p1
    archname=amd64-freebsd-thread-multi
    uname='freebsd cjg-freebsd12 12.1-release-p1 freebsd 12.1-release-p1 generic amd64 '
    config_args='-des -Dprefix=~/bin/perl-blead -Dscriptdir=~/bin/perl-blead/bin -Dusedevel -Duse64bitall -Duseithreads'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
    bincompat5005=undef
  Compiler:
    cc='cc'
    ccflags ='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_FORTIFY_SOURCE=2'
    optimize='-O2'
    cppflags='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='4.2.1 Compatible FreeBSD Clang 8.0.1 (tags/RELEASE_801/final 366581)'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags ='-pthread -Wl,-E  -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib /usr/lib/clang/8.0.1/lib /usr/lib
    libs=-lpthread -lgdbm -ldl -lm -lcrypt -lutil
    perllibs=-lpthread -ldl -lm -lcrypt -lutil
    libc=
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags=' '
    cccdlflags='-DPIC -fPIC'
    lddlflags='-shared	-L/usr/local/lib -fstack-protector-strong'


---
@INC for perl 5.31.9:
    /home/cpan/bin/perl-blead/lib/site_perl/5.31.9/amd64-freebsd-thread-multi
    /home/cpan/bin/perl-blead/lib/site_perl/5.31.9
    /home/cpan/bin/perl-blead/lib/5.31.9/amd64-freebsd-thread-multi
    /home/cpan/bin/perl-blead/lib/5.31.9

---
Environment for perl 5.31.9:
    HOME=/home/cpan
    LANG (unset)
    LANGUAGE (unset)
    LC_ALL=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/cpan/bin/perl-blead/bin:/home/cpan/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin:/home/cpan/bin
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/bash
```

**Comment by jkeenan** [comment]
> This is a bug report for perl from "Carlos Guevara" [carlos@carlosguevara.com](mailto:carlos@carlosguevara.com),
> generated with the help of perlbug 1.41 running under perl 5.31.9.
> 
> [Please describe your issue here]
> 
> BBC: Blead Breaks Sub::Quote
> 
> Please see http://matrix.cpantesters.org/?dist=Sub-Quote

Bisection points to this commit:
```
daf6caf1ef25ff48f871fa1e53adcefc11bf1d08 is the first bad commit
commit daf6caf1ef25ff48f871fa1e53adcefc11bf1d08
Author: Karl Williamson <khw@cpan.org>
Date:   Thu Jan 16 16:14:40 2020 -0700
Commit:     Karl Williamson <khw@cpan.org>
CommitDate: Thu Jan 23 15:46:56 2020 -0700


    pv_uni_display: Use common fcn; \b mnemonic
    
    This removes the (almost) duplicate code in this function to display
    mnemonics for control characters that have them.  The reason the two
    pieces of code aren't precisely the same is that the other function also
    uses \b as a mnemonic for backspace.  Using all possible mnemonics is
    desirable, so a flag is added for pv_uni_display to now use \b.  This is
    now by default enabled in double-quoted strings, but not regex patterns
    (as \b there means something quite different except in character classes).
    B.pm is changed to expect \b.
```
Bisection invocation:
```
perl Porting/bisect.pl --start=6568ef8216f93a6c31f5b549d409deeedc0991db  \
--end=fb953ad4077a4a0f7b8204f52f3ce71feca190d2 --module=Sub::Quote
```
Karl, can you take a look?

Thank you very much.
Jim Keenan

**Comment by haarg** [comment]
Condensed test case:
```
perl -e'
use B;
my $in = "\x{1b}";
utf8::upgrade($in);
my $code = B::perlstring($in);
my $out = eval $code;
print $in eq $out ? "ok\n" : "not ok\n";
'
```

**Comment by demerphq** [comment]
Fixed with 948c73696c11c6875d4a734a634cdcac79331d4c
